InternalDocs/tier2.md

```diff
@@ -0,0 +1,124 @@
+# The Tier 2 Interpreter
```

Comment:
I'm making this change, but the code does refer to tier 2 in various places so I think it needs to be mentioned, otherwise future devs will be confused.